### PR TITLE
Fix build for gazebo4

### PR DESCRIPTION
**Pull request by nkoenig**

### CODE REVIEW

**Comment by adolfo-rt** [comment]
Shouldn't this PR target `indigo-devel`?. The fix for `gazebo_ros_control_plugin.cpp` is already there.

It has already been suggested before, but we should change the repository's default branch to `indigo-devel`, especially now that indigo has been released.

**Comment by j-rivero** [comment]
I've tested the pull request under Trusty for:
- gazebo4-prerelease + Indigo
- gazebo3 + Indigo
- gazebo2 + Indigo
- gazebo + hydro

It is compiling just fine for me on all the combinations.

@adolfo-rt we are still producing ROS wrappers for gazebo new versions (gazebo3, gazebo4) in Hydro and it would help to have the fixes applied to both branches, hydro-devel and indigo-devel.

Thanks.

**Comment by hsu** [comment]
`GAZEBO_VERSION_MAJOR` should be `GAZEBO_MAJOR_VERSION`, similarly for others.

File: gazebo_ros_control/src/default_robot_hw_sim.cpp [hunk 1]
```diff
@@ -288,7 +294,11 @@ class DefaultRobotHWSim : public gazebo_ros_control::RobotHWSim
           break;
 
         case POSITION:
+#if GAZEBO_VERSION_MAJOR >= 4
```
Comment (hsu) [inline gazebo_ros_control/src/default_robot_hw_sim.cpp]: should be `GAZEBO_MAJOR_VERSION`

**Comment by adolfo-rt** [comment]
> @adolfo-rt we are still producing ROS wrappers for gazebo new versions (gazebo3, gazebo4) in Hydro and it would help to have the fixes applied to both branches, hydro-devel and indigo-devel.

OK, I was not aware of this.

**Pull request merged**